### PR TITLE
Fix for issue#63

### PR DESCRIPTION
**Pull request by terrameijar**
This addresses issue#63

Please review my work and let me know what changes/modifications need to be made.

### CODE REVIEW

**Comment by rbillings** [comment]
Fix for Issue #63, flake8 errors will need to be addressed: https://travis-ci.org/mozilla/treeherder-tests/builds/130958563

File: pages/treeherder.py [hunk 1]
```diff
@@ -22,12 +22,23 @@ class TreeherderPage(Base):
     _result_sets_locator = (By.CSS_SELECTOR, '.result-set:not(.row)')
     _unchecked_repos_links_locator = (By.CSS_SELECTOR, '#repoLabel + .dropdown-menu .dropdown-checkbox:not([checked]) + .dropdown-link')
     _unclassified_failure_count_locator = (By.ID, 'unclassified-failure-count')
+    _action_menu_dropdown_locator = (By.CSS_SELECTOR, ".th-view-content > ng-view:nth-child(1) > div:nth-child(2) > div:nth-child(1) > span:nth-child(3) > th-action-button:nth-child(4) > span:nth-child(1) > button:nth-child(1)")
```
Comment (rbillings) [inline pages/treeherder.py]: This locator is really fragile. I added this dropdown locator in another test, I would recommend that you look at this for ways to shorten your locators:
https://github.com/rbillings/treeherder-tests/blob/add_new_jobs/pages/treeherder.py#L95

Also, the locators for lines 25,26, and 27 will need to be moved to the ResultSet class.

File: pages/treeherder.py [hunk 2]
```diff
@@ -22,12 +22,23 @@ class TreeherderPage(Base):
...
+    _top_of_range_link_locator =  (By.CSS_SELECTOR,".th-view-content > ng-view:nth-child(1) > div:nth-child(2) > div:nth-child(1) > span:nth-child(3) > th-action-button:nth-child(4) > span:nth-child(1) > ul:nth-child(2) > li:nth-child(7) > a:nth-child(1)")
```
Comment (rbillings) [inline pages/treeherder.py]: Fragile locator

File: pages/treeherder.py [hunk 3]
```diff
@@ -22,12 +22,23 @@ class TreeherderPage(Base):
...
+    _bottom_of_range_link_locator = (By.CSS_SELECTOR, ".th-view-content > ng-view:nth-child(1) > div:nth-child(2) > div:nth-child(1) > span:nth-child(3) > th-action-button:nth-child(4) > span:nth-child(1) > ul:nth-child(2) > li:nth-child(8) > a:nth-child(1)")
```
Comment (rbillings) [inline pages/treeherder.py]: Fragile locator

File: pages/treeherder.py [hunk 4]
```diff
@@ -22,12 +22,23 @@ class TreeherderPage(Base):
...
 
     def wait_for_page_to_load(self):
         Wait(self.selenium, self.timeout).until(
             lambda s: self.unclassified_failure_count > 0)
         return self
 
+    def set_resultset_as_top_of_range(self):
```
Comment (rbillings) [inline pages/treeherder.py]: This method should be moved to the ResultSet class page region.

File: pages/treeherder.py [hunk 5]
```diff
@@ -22,12 +22,23 @@ class TreeherderPage(Base):
...
+        self.selenium.find_element(*self._action_menu_dropdown_locator).click()
+        self.selenium.find_element(*self._top_of_range_link_locator).click()
+
+    def set_resultset_as_bottom_of_range(self):
```
Comment (rbillings) [inline pages/treeherder.py]: This method should be moved to the ResultSet class page region.

**Comment by rbillings** [comment]
This is a great start- next steps will be fixing the locators and moving the locators and methods into the ResultSet page region.

**Comment by terrameijar** [comment]
Ok thank you, let me work on those now

**Comment by terrameijar** [comment]
@rbillings Please review again when you have a moment.

File: pages/treeherder.py [hunk 6]
```diff
@@ -93,6 +93,9 @@ class ResultSet(PageRegion):
         _datestamp_locator = (By.CSS_SELECTOR, '.result-set-title-left > span a')
         _jobs_locator = (By.CLASS_NAME, 'job-btn')
         _pin_all_jobs_locator = (By.CLASS_NAME, 'pin-all-jobs-btn')
+        _dropdown_toggle_locator = (By.CSS_SELECTOR, 'button.btn.btn-sm.btn-resultset.dropdown-toggle')
```
Comment (rbillings) [inline pages/treeherder.py]: This locator is improved, but you should use the locator here: 
https://github.com/mozilla/treeherder-tests/blob/master/pages/treeherder.py#L95

File: pages/treeherder.py [hunk 7]
```diff
@@ -93,6 +93,9 @@ class ResultSet(PageRegion):
...
+        _top_of_range_link_locator = (By.LINK_TEXT, 'Set as top of range')
```
Comment (rbillings) [inline pages/treeherder.py]: Locators should by styled using our style guide: https://wiki.mozilla.org/QA/Execution/Web_Testing/Docs/Automation/StyleGuide#Locators

This locator would use either CLASS_NAME or CSS_SELECTOR.

File: pages/treeherder.py [hunk 8]
```diff
@@ -93,6 +93,9 @@ class ResultSet(PageRegion):
...
+        _bottom_of_range_link_locator = (By.LINK_TEXT, 'Set as bottom of range')
```
Comment (rbillings) [inline pages/treeherder.py]: This should use one of the approved locators.

File: tests/test_move_to_top_or_bottom_of_range.py [hunk 9]
```diff
@@ -0,0 +1,19 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from pages.treeherder import TreeherderPage
+
+
+def test_move_to_top_of_range(base_url, selenium):
+    page = TreeherderPage(base_url, selenium).open()
+    result_set = page.ResultSet(page)
```
Comment (rbillings) [inline tests/test_move_to_top_or_bottom_of_range.py]: I can see why you did this, but it's unnecessary. You can see in this test that after opening the page, the actions can be taken by going to page.pageRegionName.method() [example: https://github.com/mozilla/treeherder-tests/blob/master/tests/test_add_new_jobs.py#L23].

File: tests/test_move_to_top_or_bottom_of_range.py [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+    result_set.set_resultset_as_top_of_range()
+    assert "&tochange=" in page.selenium.current_url
+
+
+def test_move_to_bottom_of_range(base_url, selenium):
+    page = TreeherderPage(base_url, selenium).open()
+    result_set = page.ResultSet(page)
```
Comment (rbillings) [inline tests/test_move_to_top_or_bottom_of_range.py]: Same comment as line 10.

**Comment by rbillings** [comment]
Hi @terrameijar - are you still working on this?

**Comment by terrameijar** [comment]
Hi @rbillings I would like to continue, but I had trouble figuring out how to go about the steps you outlined above. I was not able to locate using CLASS_NAME or CSS_SELECTOR. I'm new to WebDriver so I am probably missing a lot of things.

**Comment by rbillings** [comment]
@terrameijar the dropdown toggle locator has already been done, you just need to copy it from here: https://github.com/mozilla/treeherder-tests/blob/master/pages/treeherder.py#L95
For the others the styleguide linked above should help. If that is not enough, there are lots of tutorials like this one: http://www.software-testing-tutorials-automation.com/2014/01/locating-web-element-by-classname-in.html

**Comment by terrameijar** [comment]
Hi @rbillings

I'm unable to continue with this task, , all the tests are failing on my system, i don't remember chanaging anything, please take a look at the error log I get when I attempt to run the tests:
[log.txt](https://github.com/mozilla/treeherder-tests/files/390264/log.txt)

**Comment by rbillings** [comment]
@terrameijar is the branch updated with master?  It looks like it has merge conflicts. You might also try creating a new virtual environment to run the tests. If you would prefer to close this task, that is also acceptable.

**Comment by terrameijar** [comment]
@rbillings Let's close this task for now, I'll look into what the problem with my dev environment is and fix it. Hopefully there will be more tasks for me to do in the future.

**Pull request closed without merging**